### ISSUE TITLE
raidemulator can't reset battles correctly

### ISSUE THREAD

**Issue by choyike** [quisquous/cactbot#5714]
I'm trying out the raidemulator to write my own scripts. It will work fine until July 10th. When I tried to load logs yesterday, a logs file only read first fight fine, it couldn't read other fights under the same log file correctly anymore. When the battle is wipe, it reports the following error. My logs files(include olds) have same error.
![bug](https://github.com/quisquous/cactbot/assets/32677144/3cff0a4c-25b8-4564-9f30-1e9e5659f16c)

By the way，I have tried masking my own scripts file，not using OVERLAY_WS to load raidemulator(using the url:https://quisquous.github.io/cactbot/ui/raidboss/raidemulator.html), clear the browser cache, use another network and so on,
it has the same error.

**Comment by Souma-Sumire** [comment]
I've encountered the same issue as well. I've been trying to pinpoint the problem, and currently, I've noticed that there's a possibility of getting all-zero data when checking for Jobs in CombatantTracker.ts#78. I suspect this might be due to the data refresh rate.

![image](https://github.com/quisquous/cactbot/assets/33572696/45795094-fcb5-4ca2-9882-600e30ee27aa)

This change was implemented in c245ef6b45f864fc79689f6e5a87cc06cb899e6e, and when I revert to the previous commit 18665f84d9da43be54817a6cceac75611a8b8e66, the issue doesn't occur.

**Comment by Souma-Sumire** [comment]
Perhaps it's an issue with encounter segmentation, but I'm not entirely sure. Here is the partyList log:
 - 11|2023-08-30T21:02:53.0760000+08:00|8|...
- 11|2023-08-30T21:06:26.0640000+08:00|8|...
- 11|2023-08-30T21:08:07.2590000+08:00|8|...
- 11|2023-08-30T21:45:21.0270000+08:00|0|...
This is the segmented encounter:
![image](https://github.com/quisquous/cactbot/assets/33572696/f3fb6ffd-480c-4016-9de0-f8f587a53701)

**Comment by Souma-Sumire** [comment]
It seems to be caused by being split by the fflogs uploader.

**Comment by Souma-Sumire** [comment]
I have been deeply troubled by this issue, as a large number of battle records were unusable. Therefore, I have been trying to figure out how to fix this problem. 

Now, I have made some progress. 

With [this commit](https://github.com/Souma-Sumire/cactbot/commit/700d5149c5b8d92bbf849921a92b56f8592926e2), I can now successfully read battle records.

**Comment by quisquous** [comment]
Hmm! What is the bug here, ultimately? Is it that `nextState(0)` can't find a state that `nextState(encStart)` can? That is kind of surprising to me.

(cc @valarnin)

**Comment by valarnin** [comment]
Yeah, this is on my list of things to look into but it's a bit far down the list since it's resolved with a refresh when it happens.
